generate_reflection/plugin/main.lua

```diff
@@ -2,12 +2,154 @@ local HttpService = game:GetService("HttpService")
+	local version = string.split(version(), ".")
+	local major = tonumber(version[1])
+	local minor = tonumber(version[2])
+	local patch = tonumber(version[3])
+	local build = tonumber(version[4])
+
+	return {major, minor, patch, build}
+end
+
+local function set(instance, property, value)
+	instance[property] = value
+end
+
+local function get(instance, property)
+	return instance[property]
+end
+
+local function getInstance(className)
+	if classNameBlacklist[className] then
+		return nil
+	end
+
+	local canCreate, instance = pcall(Instance.new, className)
+	if canCreate then
+		return instance
+	end
+
+	local canLocate, service = pcall(game.GetService, game, className)
+	if canLocate then
+		return service
+	end
+
+	return nil
+end
+
+local function shouldSkip(propertyDescriptor)
+	local scriptability = propertyDescriptor.Scriptability
+	local isBlacklisted = propertyBlacklist[propertyDescriptor.Name]
+
+	return
+		isBlacklisted
+		or scriptability == "None"
+		or scriptability == "Custom"
+		or propertyDescriptor.Kind.Canonical == nil
+end
+
+local function getPropertyChange(instance, propertyDescriptor)
+	local propertyName = propertyDescriptor.Name
+	local readSuccess, value = pcall(get, instance, propertyName)
+	local writeSuccess = pcall(set, instance, propertyName, value)
+	local measuredScriptability
+
+	if readSuccess and writeSuccess then
+		measuredScriptability = "ReadWrite"
+	elseif readSuccess then
+		measuredScriptability = "Read"
+	else
+		-- TODO: Are there any properties that are writable, but not
+		-- readable?
+		measuredScriptability = "None"
+	end
+
+	if measuredScriptability ~= propertyDescriptor.Scriptability then
+		return {
+			Scriptability = measuredScriptability,
+		}
+	else
+		return nil
+	end
+end
+
+local function getPropertyPatches(classes)
+	local propertyChanges = {}
+
+	for className, classDescriptor in pairs(classes) do
+		local instance = getInstance(className)
+
+		if instance == nil then
+			continue
+		end
+
+		local changes = {}
+
+		for propertyName, propertyDescriptor in pairs(classDescriptor.Properties) do
+			if shouldSkip(propertyDescriptor) then
+				continue
+			end
+
+			changes[propertyName] = getPropertyChange(instance, propertyDescriptor)
+		end
+
+		if next(changes) then
+			propertyChanges[className] = changes
+		end
+
+		local superclassName = classDescriptor.Superclass
+
+		while superclassName ~= nil do
+			if propertyChanges[superclassName] ~= nil then
+				break
+			end
+
+			local superclassDescriptor = classes[superclassName]
+			local superclassChanges = {}
+
+			for propertyName, propertyDescriptor in pairs(superclassDescriptor.Properties) do
+				if shouldSkip(propertyDescriptor) then
+					continue
+				end
+
+				superclassChanges[propertyName] = getPropertyChange(instance, propertyDescriptor)
+			end
+
+			if next(superclassChanges) then
+				propertyChanges[superclassName] = superclassChanges
+			end
+
+			superclassName = superclassDescriptor.Superclass
+		end
```

Comment:
Thought of a possible issue here: if we can't read/write superclass properties on a particular subclass, but *can* on other subclasses, *and* we happen measure the former kind of subclass first, then these properties' measured scriptability will be wrong! AFAIK this is only possible with deprecated classes like `Speaker` that subclass `Instance` - we currently ignore these properties across the board, thus avoiding the issue. Should we worry about this?